src/lib.rs

```diff
@@ -83,25 +94,101 @@ fn get_output_mode(args: &Args) -> OutputMode {
     }
 }
 
+struct MaybeInitializedCaptureIndex(AtomicU32);
+
+impl MaybeInitializedCaptureIndex {
+    const UNINITIALIZED: u32 = u32::MAX;
+
+    const FAILED: u32 = u32::MAX - 1;
```

Comment:
These `u32::MAX`/`u32::MAX - 1`'s (while encapsulated) were seeming a bit "strewn about" so added named constants for them